### PR TITLE
[2.6 Backport]Add hardening config for rancher provisioned rke2 clusters

### PR DESCRIPTION
**Pull request by anupama2501**
## Problem
Currently, the Go test framework allows for provisioning RKE2 custom clusters. By default, these are non-hardened clusters. In realistic customer environments, hardened clusters are at the forefront, so we need to be able to support this in our testing. This PR addresses this problem by providing the option to harden the custom RKE2 clusters.

## Solution

This is a multi-step solution, please find a summary of the changes incorporated:
- There needed to be an option available to select whether or not the user wishes to harden the cluster. This is done by adding a new Hardened bool type in the config.go struct. Inside the user's cattle-config.yaml, they will have to define a new hardened value and set it to true or false.
- In the custom_clusters_test.go file, new functions HardenNodes, HardenK3SRKE2ClusterConfig and UpdateK3SRKE2Cluster are referenced. We will break each down in the subsequent steps, consequently.
- HardenNodes is found in the new hardening/harden_nodes.go. The point of the file is it will check the new Hardened value as mentioned in step 1. If user says yes, harden the node following appropriate hardening steps.
- Once step 3 is finished, HardenK3SRKE2ClusterConfig will update the cluster's configuration to harden the cluster.
- Once step 4 is finished, UpdateK3SRKE2Cluster will call upon the Steve client's update function to complete the changes to actually harden the K3s custom cluster.

### CODE REVIEW

**Review by markusewalker — COMMENTED** [review]
Overall, LGTM! The same comments I had in the forwardport are the same ones that I have here.

File: tests/v2/validation/provisioning/rke2/custom_cluster_test.go
Comment (markusewalker) [inline tests/v2/validation/provisioning/rke2/custom_cluster_test.go:173]: Same last nit-pick I had in the forwardport. Can we make sure the code is more readable, just like how you have in the dynamic version? Everything else lgtm!

**Review by markusewalker — DISMISSED** [review]

**Review by markusewalker — APPROVED** [review]
LGTM

**Review by igomez06 — DISMISSED** [review]
LGTM

**Review by slickwarren — APPROVED** [review]

**Comment by igomez06** [comment]
This is has slipped through the cracks, Anu could you rebase so we can get this in? Thank you!

**Review by caliskanugur — COMMENTED** [review]
Looks good to me, like @igomez06 mentioned, seems like it only needs a rebase

File: tests/v2/validation/provisioning/rke2/custom_cluster_test.go
Comment (igomez06) [reply tests/v2/validation/provisioning/rke2/custom_cluster_test.go:145]: Thanks Ugur. Yeah it still looks like it not fully updated.

**Review by markusewalker — APPROVED** [review]
LGTM, the last rebase seems to have updated the `clusters.go` file appropriately.

**Review by igomez06 — APPROVED** [review]
It looks good now. Just waiting on a Jenkins job before the merge. Thank you!

**Pull request merged**
